rta/__init__.py

```diff
@@ -3,42 +3,56 @@
 # 2.0; you may not use this file except in compliance with the Elastic License
 # 2.0.
 
-import glob
 import importlib
-import os
+from pathlib import Path
+from typing import Dict, List
 
 from . import common
 
-CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+CURRENT_DIR = Path(__file__).parent.absolute()
 
 
-def get_ttp_list(os_types=None):
-    scripts = []
-    if os_types and not isinstance(os_types, (list, tuple)):
-        os_types = [os_types]
+def get_available_tests(
+    print_list: bool = False, os_filter: str = None
+) -> (List, List[Dict]):
+    """Get a list of available tests."""
+    test_names = []
+    test_metadata = []
 
-    for script in sorted(glob.glob(os.path.join(CURRENT_DIR, "*.py"))):
-        base_name, _ = os.path.splitext(os.path.basename(script))
-        if base_name not in ("common", "main") and not base_name.startswith("_"):
-            if os_types:
-                # Import it and skip it if it's not supported
-                importlib.import_module(__name__ + "." + base_name)
-                if not any(base_name in common.OS_MAPPING[os_type] for os_type in os_types):
-                    continue
+    for file in CURRENT_DIR.rglob("*.py"):
 
-            scripts.append(script)
+        if file.stem not in ("common", "main") and not file.stem.startswith("_"):
+            module = importlib.import_module(f"rta.{file.stem}")
 
-    return scripts
+            if os_filter and os_filter not in module.PLATFORMS and os_filter != "all":
+                continue
+            test_names.append(file.stem)
+            test_metadata.append(
+                {
```

Comment:
Can you create a dataclass in this file and import it into the RTA's (no validation). This will help with typing

something like

```python
@dataclass
class RtaMetadata:
    """Metadata associated with all RTAs."""
    uuid: str
    platforms: List[str]

    path: Path = field()  # ...autogenerate based on the importing file (the RTA)
    name: str = field()  # ...autogenerate based on the importing file (the RTA)
    endpoint: Optional[List[dict]] = None
    siem: Optional[List[dict]] = None
    techniques: Optional[List[str]] = None
```